src/js/launcher.js

```diff
@@ -0,0 +1,115 @@
+"use strict";
+var fluid  = require("infusion");
+var gpii   = fluid.registerNamespace("gpii");
+var kettle = fluid.registerNamespace("kettle");
+
+fluid.require("%kettle/lib/KettleConfigLoader.js");
+
+var yargs   = require("yargs");
+var path    = require("path");
+var process = require("process");
+
+// A global resolver that can be used to pull raw environment variables from configuration blocks.
+fluid.defaults("gpii.launcher.resolver", {
+    gradeNames: ["fluid.component", "fluid.resolveRootSingle"],
+    singleRootType: "gpii.launcher.resolver",
+    members: {
+        env:  process.env
+    }
+});
+
+// Create an instance of gpii.resolvers.env
+fluid.construct("gpii_launcher_resolver", {
+    type: "gpii.launcher.resolver"
+});
+
+fluid.registerNamespace("gpii.launcher");
+
+/**
+ *
+ * Merge options from (in order of precedence):
+ *
+ * 1. An options file, if supplied.
+ * 2. Environment variables.
+ * 3. Command line parameters.
+ *
+ * @param that - The `gpii.launcher` component itself.
+ *
+ */
+gpii.launcher.launchComponent = function (that) {
+    fluid.each(that.options.yargsOptions, function (fnArgs, fnName) {
+        yargs[fnName].apply(yargs, fluid.makeArray(fnArgs));
+    });
+
+    var args = fluid.copy(yargs.argv);
+    yargs.reset();
+
+    var paramAndEnvironmentOptions = that.options.filterKeys ? that.filterKeys(args) : args;
+
+    var fullPath   = gpii.launcher.resolvePath (args.optionsFile);
```

Comment:
The space was not picked up by the linting config, otherwise I wouldn't have been able to commit the file.

I will need a bit to consider the arrays of optionsFile options and other concerns, and will comment further once I have ideas on how to proceed.